### PR TITLE
chore(content-manager): convert the editview to typescript

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* converts the editview and associated components to TS

### Why is it needed?

* TS TS TS TS

### Related issue(s)/PR(s)

* resolves CONTENT-1944

### CODE REVIEW

**Comment by joshuaellis** [comment]
I won't build because we need to merge & release https://github.com/strapi/design-system/pull/1487 first.

**Comment by joshuaellis** [comment]
I still need to fix the E2E tests, but as you can see this was a very big job, so i think it's fine to start reviewing the code.

File: packages/core/admin/admin/src/content-manager/components/DynamicZone/ComponentCategory.tsx [hunk 1]
```diff
@@ -1,28 +1,46 @@
-import React from 'react';
+import * as React from 'react';
```
Comment (Marc-Roig) [inline packages/core/admin/admin/src/content-manager/components/DynamicZone/ComponentCategory.tsx:1]: is there a difference between this and import React from 'react'?

**Comment by soupette** [comment]
Wow that's a huge work and improvement!

**Review by Marc-Roig — APPROVED** [review]
I don't see any apparent issue in the EditView.
🔥

**Pull request merged**